app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+    };
+
+    const getChannels = useCallback(debounce(async () => {
+        if (next && !loading && !term) {
+            setCurrentPage(currentPage + 1);
+            const {channels: channelData} = await fetchChannels(serverUrl, currentTeamId, currentPage);
+
+            if (channelData && channelData.length > 0) {
+                loadedChannels([...integrationData as Channel[], ...channelData]);
+            }
+        }
+    }, 100), [integrationData]);
+
+    const getProfiles = useCallback(debounce(async () => {
+        if (next && !loading && !term) {
+            setCurrentPage(currentPage + 1);
+            const {users: userData} = await fetchProfiles(serverUrl, currentPage);
+
+            if (userData && userData.length > 0) {
+                loadedProfiles([...integrationData as UserProfile[], ...userData]);
+            }
+        }
+    }, 100), [integrationData]);
+
+    const getDynamicOptionsLocally = () => {
+        if (!loading && !term) {
+            searchDynamicOptions('');
+        }
+    };
+
+    const loadedChannels = (channels: Channel[]) => {
+        if (channels && !channels.length) {
+            setNext(false);
+        }
+        setIntegrationData(channels);
+    };
+
+    const loadedProfiles = (profiles: UserProfile[]) => {
+        if (profiles && !profiles.length) {
+            setNext(false);
+        }
+        setIntegrationData(profiles);
+    };
+
+    const loadMore = () => {
+        setLoading(true);
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            getChannels();
+        }
+
+        setLoading(false);
+
+        // dynamic options are not paged so are not reloaded on scroll
+    };
+
+    const searchChannels = async (searchTerm: string) => {
+        const isSearch = true;
+        const {channels: receivedChannels} = await searchChannelsRemote(serverUrl, searchTerm, currentTeamId, isSearch);
+
+        if (receivedChannels) {
+            setSearchResults(receivedChannels);
+        }
+    };
+
+    const searchProfiles = async (searchTerm: string) => {
+        const {data: userData} = await searchProfilesRemote(serverUrl, searchTerm.toLowerCase(), {team_id: currentTeamId, allow_inactive: true});
+
+        if (userData) {
+            setSearchResults(userData);
+        }
+    };
+
+    const searchDynamicOptions = useCallback((searchTerm = '') => {
+        if (options) {
+            setIntegrationData(options);
+        }
+
+        if (!getDynamicOptions) {
+            return;
+        }
+
+        getDynamicOptions(searchTerm.toLowerCase()).then((results: DialogOption[]) => {
+            const searchData = results || [];
+
+            if (searchTerm) {
+                setSearchResults(searchData);
+            } else {
+                setIntegrationData(searchData);
+            }
+        });
+    }, [term, integrationData, searchResults]);
```

Comment:
This depends on `options` and `getDynamicOptions`, but I see no reason to depend on `term`, `integrationData` nor `searchResults`.